src/test/java/org/jenkinsci/plugins/gitclient/JcascTest.java

```diff
@@ -0,0 +1,81 @@
+
+package org.jenkinsci.plugins.gitclient;
+
+import hudson.plugins.git.GitTool;
+import hudson.tools.BatchCommandInstaller;
+import hudson.tools.InstallSourceProperty;
+import hudson.tools.ToolDescriptor;
+import hudson.tools.ToolInstallation;
+import hudson.tools.ToolProperty;
+import hudson.tools.ToolPropertyDescriptor;
+import hudson.tools.ZipExtractionInstaller;
+import hudson.util.DescribableList;
+import io.jenkins.plugins.casc.misc.RoundTripAbstractTest;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.beans.HasPropertyWithValue.hasProperty;
+import static org.hamcrest.collection.IsArrayContainingInAnyOrder.arrayContainingInAnyOrder;
+import static org.hamcrest.collection.IsArrayWithSize.arrayWithSize;
+import static org.hamcrest.collection.IsIterableWithSize.iterableWithSize;
+import static org.hamcrest.core.AllOf.allOf;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.Assert.assertThat;
+
+public class JcascTest extends RoundTripAbstractTest {
+    @Override
+    protected void assertConfiguredAsExpected(RestartableJenkinsRule restartableJenkinsRule, String s) {
+        final ToolDescriptor descriptor = (ToolDescriptor) restartableJenkinsRule.j.jenkins.getDescriptor(GitTool.class);
+        final ToolInstallation[] installations = descriptor.getInstallations();
+        assertThat(installations, arrayWithSize(4));
+        assertThat(installations, arrayContainingInAnyOrder(
+                allOf(
+                        instanceOf(JGitTool.class),
+                        hasProperty("name", equalTo(JGitTool.MAGIC_EXENAME))
+                ),
+                allOf(
+                        instanceOf(JGitApacheTool.class),
+                        hasProperty("name", equalTo(JGitApacheTool.MAGIC_EXENAME))
+                ),
+                allOf(
+                        instanceOf(GitTool.class),
+                        not(instanceOf(JGitTool.class)),
+                        not(instanceOf(JGitApacheTool.class)),
+                        hasProperty("name", equalTo("Default")),
+                        hasProperty("home", equalTo("git"))
+                ),
+                allOf(
+                        instanceOf(GitTool.class),
+                        not(instanceOf(JGitTool.class)),
+                        not(instanceOf(JGitApacheTool.class)),
+                        hasProperty("name", equalTo("optional")),
+                        hasProperty("home", equalTo("/opt/git/git"))
+                ))
```

Comment:
How are these two different?

Reply:
The first one is the default one added by GitTool during startup if there is no configuration. The second one has been manually added and also has tool properties configured.

Reply:
Yes, I got that but the "filtering" is the same: instancess of GitTool and not of JGitTool or JGitApacheTool, so I don't see why in one case one is returned and another one in the other case.

Reply:
```
hasProperty("name", equalTo("Default")),
hasProperty("home", equalTo("git"))
```
vs.
```
hasProperty("name", equalTo("optional")),
hasProperty("home", equalTo("/opt/git/git"))
```

Reply:
Ahhhhh the assert is checking the order in the array. Fine, I got it now, thanks!

Reply:
It's checking that everything defined is in the array, but ignores the specific order or the items :)